docs/source/developer_guide/development-workflow.md

```diff
@@ -128,7 +128,13 @@ You can install Elyra using a local build of @elyra/pipeline-editor with:
 make clean dev-link install
 ```
 
-### Incremental Development
+### Back-end Development
```

Comment:
I would describe this in more details similar to the one below (--dev-lint) like when to use, what it does, etc